client-ci-analysis/ci-testfailure-analyzer/Program.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.IO;
+using System.Net.Http.Headers;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using ci_testfailure_analyzer.Models.AzDO;
+using System.Linq;
+
+namespace ci_testfailure_analyzer
+{
+    internal class Program
+    {
+        static async Task Main(string[] args)
+        {
+            if (args.Length == 0)
+            {
+                Console.WriteLine("Expected at least 1 argument, got " + args.Length);
+                return;
+            }
+
+            if (File.Exists(args[0]))
+            {
+                Console.WriteLine("Expected '{0}' to be a directory, but found a file", args[0]);
+                return;
+            }
+
+            var cache = new DirectoryInfo(args[0]);
+            if (!cache.Exists)
+            {
+                cache.Create();
+            }
+
+            var buildDefinition = 8118; //private build. Use 8117 for official, 14219 for trusted build.
+
+            if (args.Length > 1 && int.TryParse(args[1], out buildDefinition))
+            {}
+
+            Console.WriteLine($"Cache path: {cache.FullName}");
+            Console.WriteLine($"CI pipeline Build Definition: {buildDefinition}");
+
+            var accountName = Environment.GetEnvironmentVariable("AzDO_ACCOUNT");
+
+            // You can get bearer token from web browser use here, had no time to fix automatically getting it.
```

Comment:
Somehow PAT doesn't work for POST request to CI pipeline end point, had no time to address, I'll take a look later.